nilearn/maskers/nifti_labels_masker.py

```diff
@@ -594,42 +607,7 @@ def transform_single_imgs(self, imgs, confounds=None, sample_mask=None):
                 imgs_,
                 self._resampled_labels_img_,
             ):
-                if self.verbose > 0:
-                    print("Resampling labels")
-                labels_before_resampling = set(
-                    np.unique(
-                        _utils.niimg.safe_get_data(
-                            self._resampled_labels_img_,
-                        )
-                    )
-                )
-                self._resampled_labels_img_ = self._cache(
-                    image.resample_img, func_memory_level=2
-                )(
-                    self.labels_img_,
-                    interpolation="nearest",
-                    target_shape=imgs_.shape[:3],
-                    target_affine=imgs_.affine,
-                )
-                labels_after_resampling = set(
-                    np.unique(
-                        _utils.niimg.safe_get_data(
-                            self._resampled_labels_img_,
-                        )
-                    )
-                )
-                labels_diff = labels_before_resampling.difference(
-                    labels_after_resampling
-                )
-                if len(labels_diff) > 0:
-                    warnings.warn(
-                        "After resampling the label image to the "
-                        "data image, the following labels were "
-                        f"removed: {labels_diff}. "
-                        "Label image only contains "
-                        f"{len(labels_after_resampling)} labels "
-                        "(including background)."
-                    )
+                self._resample_labels(imgs_)
```

Comment:
this refactoring is unrelated but this `transform_single_imgs` was getting way too long...